presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1955,15 +1957,15 @@ public void dropRole(ConnectorSession session, String role)
     }
 
     @Override
-    public void grantRoles(ConnectorSession session, Set<String> roles, Set<PrestoPrincipal> grantees, boolean withAdminOption, Optional<PrestoPrincipal> grantor)
+    public void grantRoles(ConnectorSession session, Set<Name> roles, Set<PrestoPrincipal> grantees, boolean withAdminOption, Optional<PrestoPrincipal> grantor)
     {
-        metastore.grantRoles(roles, HivePrincipal.from(grantees), withAdminOption, grantor.map(HivePrincipal::from).orElse(new HivePrincipal(USER, session.getUser())));
+        metastore.grantRoles(roles.stream().map(Name::getLegacyName).collect(toImmutableSet()), HivePrincipal.from(grantees), withAdminOption, grantor.map(HivePrincipal::from).orElse(new HivePrincipal(USER, session.getUser())));
```

Comment:
This is hard to read. I'd format as:

```java
metastore.grantRoles(
    roles.stream()
        .map(Name::getLegacyName)
        .collect(toImmutableSet()), 
    HivePrincipal.from(grantees), 
    withAdminOption, 
    grantor.map(HivePrincipal::from).orElse(new HivePrincipal(USER, session.getUser())));
```

or use temporary variable for `roles.stream().map(Name::getLegacyName).collect(toImmutableSet())`